framework/src/timesteppers/CompositionDT.C

```diff
@@ -60,6 +64,79 @@ CompositionDT::CompositionDT(const InputParameters & parameters)
           "lower_bound", "Failed to find a timestepper with the name '", time_stepper_name, "'");
 }
 
+void
+CompositionDT::init()
+{
+  const auto time_steppers = getTimeSteppers();
+  for (auto & ts : time_steppers)
+    if (ts != this)
+      ts->init();
+}
```

Comment:
Make a private templated method:

```
template <typename Lambda>
void CompositionDT::actOnTimeSteppers(Lambda & act)
{
  for (auto & ts : getTimeSteppers())
    if (ts != this)
      act(*ts);
}
```

You can then replace all of these calls with something like:

```
void
CompositionDT::preExecute()
{
  actOnTimeSteppers([](auto & ts) { ts.preExecute(); });
}
```